lib/membrane/core/element/lifecycle_controller.ex

```diff
@@ -66,14 +72,22 @@ defmodule Membrane.Core.Element.LifecycleController do
   def handle_shutdown(reason, state) do
     playback_state = state.playback.state
 
-    if playback_state == :terminating do
-      Membrane.Logger.debug("Terminating element, reason: #{inspect(reason)}")
-    else
-      Membrane.Logger.warn("""
-      Terminating element possibly not prepared for termination as it was in state #{inspect(playback_state)}.
-      Reason: #{inspect(reason)}",
-      State: #{inspect(state, pretty: true)}
-      """)
+    cond do
+      playback_state == :terminating ->
+        Membrane.Logger.debug("Terminating element, reason: #{inspect(reason)}")
+
+      reason in @safe_shutdown_reasons ->
+        Membrane.Logger.warn("""
+        Terminating element possibly not prepared for termination as it was in state #{inspect(playback_state)}.
+        Reason: #{inspect(reason)}"
+        """)
```

Comment:
I'd make this a debug too

```suggestion
        Membrane.Logger.debug("""
        Terminating element possibly not prepared for termination as it was in state #{inspect(playback_state)}.
        Reason: #{inspect(reason)}"
        """)
```